File: src/dto/organization.py
from pydantic import BaseModel

from src.dto.workspace import WorkspaceSubElement
from src.models.common import UserAccess


class CreateOrganizationPayload(BaseModel):
    name: str


class UpdateOrganizationPayload(BaseModel):
    organization_id: int
    name: str


class AddUserToOrganizationElement(BaseModel):
    user_id: int
    access: UserAccess


class AddUserToOrganizationPayload(BaseModel):
    organization_id: int
    users: list[AddUserToOrganizationElement]


class UserOrganizationWithWorkspaces(BaseModel):
    organization_id: int
    name: str
    access: UserAccess
    workspaces: list[WorkspaceSubElement]

File: front/src/stores/models/organization.ts
import type { UserAccess } from "./common";
import type { WorkspaceModel } from "./workspace";

export interface OrganizationModel {
  organization_id: number,
  name: string,
  access: UserAccess,
  workspaces: WorkspaceModel[],
}

export interface OrganizationUpdateResponse  {
  id: number,
  name: string,
}

export interface OrganizationUpdateModel {
  organization_id: number,
  name: string,
}
